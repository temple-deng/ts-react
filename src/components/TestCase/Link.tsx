type LinkProps = {
  href: string;
  name: string;
};

function Link(props: LinkProps) {
  return (
    <a href={props.href}>{props.name}</a>
  );
}

export default Link;