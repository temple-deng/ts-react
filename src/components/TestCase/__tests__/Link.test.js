import renderer from 'react-test-renderer';
import Link from '../Link';

it.skip('link render', () => {
  const tree = renderer
    .create(<Link href="https://www.qq.com" name="Facebook" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it.skip('link render inline snapshot', () => {
  const tree = renderer
    .create(<Link href="https://www.qq.com" name="Facebook" />)
    .toJSON();

  expect(tree).toMatchInlineSnapshot(`
    <a
      href="https://www.qq.com"
    >
      Facebook
    </a>
  `);
});
