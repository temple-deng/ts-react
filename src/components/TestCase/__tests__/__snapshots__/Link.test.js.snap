// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link render 1`] = `
<a
  href="https://www.qq.com"
>
  Facebook
</a>
`;
